### PR TITLE
Eliminate Warning when configuring YAML

### LINKED ISSUE #15641: Changes to DataSourcePoolCreator cause warnings in spring.shardingsphere.datasource configuration
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

5.1.0

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

`ShardingSphere-JDBC`

### Expected behavior

Everything configured in YAML should work fine.

### Actual behavior

as the picture shows.

7efe1e3240b9552a55b0e4fe46c0b237ef70caee_2_1035x493

### Reason analyze (If you can)

In https://github.com/apache/shardingsphere/pull/15249 , used `org.apache.shardingsphere.infra.datasource.pool.creator.DataSourcePoolCreator` instead of `org.apache .shardingsphere.spring.boot.util.DataSourceUtil` , and removed the following json segment.
```json
{
   "properties": [
     {
       "name": "spring.shardingsphere.datasource",
       "type": "java.util.Map<java.lang.String,java.util.Map<java.lang.String,java.lang.Object>>",
       "sourceType": "org.apache.shardingsphere.spring.boot.util.DataSourceUtil"
     }
   ]
}
```
This leads to a unified warning for the configuration under `spring.shardingsphere.datasource`, which is similar to `Cannot resolve configuration property 'spring.shardingsphere.datasource.shardingmaster.jdbc-url'`

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

Non-runtime operations.

### Example codes for reproduce this issue (such as a github link).

Reference https://community.sphere-ex.com/t/topic/786 .

### PR DESCRIPTION
**Pull request by linghengqian**
Fixes #15641.

Changes proposed in this pull request:
- Eliminate Warning when configuring YAML

### CODE REVIEW

**Comment by linghengqian** [comment]
- This appears to be a CI/CD issue due to timeouts. But I have no experience with Spring 4.
- image

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 1]
```diff
@@ -70,6 +70,10 @@
       "type": "java.lang.String",
       "sourceType": "org.apache.shardingsphere.spring.boot.ShardingSphereAutoConfiguration"
     },
+    {
+      "name": "spring.shardingsphere.datasource",
+      "type": "java.util.Map<java.lang.String,java.util.Map<java.lang.String,java.lang.Object>>"
```
Comment (totalo) [inline shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json:75]: Please add blank after `,`.

**Review by totalo — CHANGES_REQUESTED** [review]

File: shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json (hunk 1 above)
Comment (totalo) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json:75]: Format of generics in Map.

Comment (linghengqian) [reply shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: processed.

**Comment by totalo** [comment]
Good job! Thanks.@linghengqian

**Review by totalo — APPROVED** [review]
Good job! Thanks.@linghengqian

**Pull request merged**